src/handlers/store-standardised-invoices/handler.ts

```diff
@@ -24,23 +26,52 @@ export const handler = async (event: SQSEvent): Promise<Response> => {
   if (rawInvoiceBucket === undefined || rawInvoiceBucket.length === 0)
     throw new Error("Raw invoice bucket not set.");
 
+  for (const records of event.Records) {
+    if (!records || !records.body) {
+      throw new Error("Record contains no body");
+    }
+  }
+
+  let recordBodies: StandardisedLineItem[] = [];
+  for (const records of event.Records) {
+    const recordList: StandardisedLineItem[] = JSON.parse(records.body);
+    recordBodies = recordBodies.concat(recordList);
+  }
   const response: Response = { batchItemFailures: [] };
 
-  const promises = event.Records.map(async (record) => {
+  const promises = recordBodies.map(async (body) => {
     try {
       await storeLineItem(
-        record,
+        body,
         destinationBucket,
         destinationFolder,
-        archiveFolder,
-        rawInvoiceBucket
+        archiveFolder
       );
     } catch (error) {
-      logger.error("Handler failure", { error });
-      response.batchItemFailures.push({ itemIdentifier: record.messageId });
+      logger.error(`Handler failure for ${JSON.stringify(body)}`, { error });
+      response.batchItemFailures.push({
+        itemIdentifier: JSON.stringify(body),
+      });
     }
   });
 
   await Promise.all(promises);
+
+  if (response.batchItemFailures.length === 0) {
```

Comment:
AWS::Serverless::Api

Reply:
not sure I follow, if some have succeeded and some have failed we don't want to move the file to the success folder we would want to reprocess it

Reply:
This  call is for moving the invoice file from the bucket to /successful. The files we create per line for further processing are saved as part of store-line-item.ts (putTextS3)